helper_funcs.inc.php

```diff
@@ -242,9 +315,29 @@ function pwg_privacy_verify_access ($img_id, $req_path) {
 		return pwg_privacy_generate_derivative($element_info, $req_path);
 	}
 
+		if (pwg_privacy_is_alternate_format_file($path,$req_path)) {
+		// this is a valid alternative format file.
+		if ($user['enabled_high']) {
+			return $req_path;
+		}
+		return pwg_privacy_error('User has no high res access ' . $img_id);
+	}
+
 	return pwg_privacy_error("Could not validate path ($req_path) actually belong to image ($img_id) ($base_file)");
 }
 
+function pwg_privacy_is_alternate_format_file ($original_file_path, $req_path) {
+	// the the base path for the original file
+	$base_img_path = preg_replace('#(.*/)[^/]*\.[^.]*$#','$1',$original_file_path);
+	// get the filename without extension of the original file
+	$base_img_name = preg_replace('#.*/([^/]*)\.[^.]*$#','$1',$original_file_path);
```

Comment:
All this regex looks a bit comple. Any reason not to use `pathinfo` here instead ?

Reply:
not specifically. I know regexp, I don't know php :-)

Reply:
So it's probably better to see if pathinfo info gives the desired results and use it instead